browser/net/ipfs_redirect_network_delegate_helper.cc

```diff
@@ -67,7 +66,9 @@ int OnHeadersReceived_IPFSRedirectWork(
   if (ctx->ipfs_auto_fallback && !api_gateway && response_headers &&
       response_headers->GetNormalizedHeader("x-ipfs-path", &ipfs_path) &&
       // Make sure we don't infinite redirect
-      !ctx->request_url.DomainIs(ctx->ipfs_gateway_url.host())) {
+      !ctx->request_url.DomainIs(ctx->ipfs_gateway_url.host()) &&
+      // Do not redirect if the page is not ipfs/ipns
+      IsIPFSScheme(ctx->tab_origin)) {
```

Comment:
i support this change but actually when we originally discussed it, i mentioned that IPFS resource loads should be blocked if any of the parent frames is non-IPFS, not just the top-level frame. 

so in the case where `ipfs://A` has an iframe to `https://B` which then has an iframe to `ipfs://C`, the `ipfs://C` load should be blocked. @bbondy is this actually the case?

if so then redirects should also be blocked in that case

Reply:
i think the blocking is implemented here? https://github.com/brave/brave-core/blob/master/browser/net/ipfs_redirect_network_delegate_helper.cc#L41-L43

if `ctx->initiator_url` is the thing described in https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/mojom/url_request.mojom;l=139-148;drc=1f97794f93e8a29186ed208d22027a43608446cb;bpv=1;bpt=1?q=url_request.mojom&ss=chromium%2Fchromium%2Fsrc, then it would be the URL of the parent frame or worker, not necessarily the tab origin (which is fine)

Reply:
done, thanks